lib/pod-names.js

```diff
@@ -0,0 +1,32 @@
+var Plugin = require('broccoli-plugin');
+var walkSync = require('walk-sync');
+var fs = require('fs');
+var path = require('path');
+var componentNames = require('./component-names.js');
+
+module.exports = PodNames;
+
+PodNames.prototype = Object.create(Plugin.prototype);
+PodNames.prototype.constructor = PodNames;
+function PodNames(inputNode, options) {
+  options = options || {};
+  Plugin.call(this, [inputNode], {
```

Comment:
basically in build, you want to calculate the operations required to "update" the persistent output directory.

fs-tree is a module written to help compare the current input dir, with the last, which allows one to write functions like [this one](https://github.com/stefanpenner/broccoli-persistent-filter/blob/master/index.js#L62-L91) which are extremely efficient.

I am sorry the current public API for these types of performant plugins is so shitty, I haven't gotten around to implementing my higher level + more digestable ideas.